### ISSUE TITLE
Word Suggestions Not Working?

### ISSUE THREAD

**Issue by seniorm0ment** [florisboard/florisboard#2307]
Word suggestions don't seem to work. Is it still in progress?
They show as the word with a number for example

Example1 Example2 Example3 Example4 etc 

Florisboard Beta Fdroid (0.4.0-alpha04)

**Comment by nettnikl** [comment]
Yes, this is currently being reworked. If you want to help developement and to get this feature to release, you can compile the latest developement version and hunt for bugs. There you can get proper suggestions, too.

**Comment by Doc-4** [comment]
You might consider pushing an unstable or alpha build for testing if suggestion is at the point that it needs it, I wasn't aware the current codebase had features working that weren't in the last beta or from some time ago. Anyone only watching the FDroid page without a brain might think nothing has happened for a while. When I have time I'll compile this and look for bugs too

**Comment by marsmathis** [comment]
@nettnikl sorry for hijacking this issue, but I just built the latest version and there’s no word suggestions in the debug apk. Did I choose a wrong branch or what could be the reason?

**Comment by nettnikl** [comment]
No worries. I'm on 0.4.0-debug-ce558c2f (90), however slightly dirty working dir. Will check later if I can get the current dev to work.

**Comment by Doc-4** [comment]
I tried building this alongside nlp in Android Studio on Windows, but was told it's not supported. I tried the Gradle.bat script which said it built something, but I could not find exactly what it did. Are there build instructions? I could troubleshoot better than this, but when it comes to something like this, it's easier to ask. I usually don't build Android apps from source, so maybe I've made an obvious mistake

**Comment by nettnikl** [comment]
Hm, sorry, windows is not my strong side - no clue what could be the issue there.
Are you on implement-latin-nlp-provider-2 or on the master branch?

**Comment by Doc-4** [comment]
> Hm, sorry, windows is not my strong side - no clue what could be the issue there. Are you on implement-latin-nlp-provider-2 or on the master branch?

Cloning the repo with git clone left the NLP folder empty, so I went to wherever the folder links to and cloned that repo and put it where it was supposed to go in the file structure. I'm not sure which that makes it. What environment do you typically build on? I'll set up a WSL2 instance and do it there.

**Comment by nettnikl** [comment]
Using Manjaro linux. But for build issues with NLP, a better adress is #2218, so this thread is not spammed and everyone looking for information about their own failing build finds it at the right place.

**Comment by Doc-4** [comment]
Thanks. I've since built it, signed it and have word suggestions working. Works very well so far, daily driving it on a pixel 8 pro running GrapheneOS. I'll submit issues if I find any bugs. Looking forward to the eventual addition of AutoCorrect.

**Comment by JetpackJackson** [comment]
@Doc-4 sorry for the ping, but I would like to build florisboard with suggestions as well, but I can't find build instructions, could you point me to them since you seem to have either found them or figured it out? I've never built anything for Android so I'm not sure what I should be looking for lol. Thanks in advance.

**Comment by Doc-4** [comment]
I personally set everything up within WSL2 but it's probably easier to just install Android Studio on whatever platform you're on right now. Then do the following:
Open terminal somewhere and run:
git clone --recursive https://github.com/florisboard/florisboard
Enter the new folder you just made. If you're on Windows, open terminal and run
.\gradlew assembleDebug
On linux:
./gradlew assembleDebug
Debug APK is generated under a few folders, install it. You could also build a release one, but that requires you to sign it yourself. I would just use this for now.

**Comment by Doc-4** [comment]
Some warnings in the current aren't up to date and have no bearing on the current functionality of the app built from source. 

I rebuilt it recently hoping to update, the plugin section isn't where it was in the last revision, and I don't follow active development enough to know why, so I downgraded to the one I was already using. In terms of just building apps, there's not much else to know or change. If you have further questions and they're relevant to typing suggestions functionality this is the right place

**Comment by JetpackJackson** [comment]
Alright, good to know. Should I try signing the app to see if that allows suggestions to be used? Do I need to build from a different git branch/repo? Cause right now with the debug apk that I built it still shows suggestions like "next1, next2, etc."

**Comment by Doc-4** [comment]
Oh, it might fix it. Just replace assembleDebug with assembleRelease. I signed it lazily using [App Manager](https://github.com/MuntashirAkon/AppManager) since it lets you create a signing key and sign apps while installing them.

**Comment by JetpackJackson** [comment]
Alright. I'm currently trying to build from the `implement-latin-nlp-provider-2` branch, once it finishes I'll see if it works.

**Comment by itsmartashub** [comment]
Great, good luck. I would appreciate it if you could let me know if you encounter any problems during the build process, and whether you succeed at the end

**Comment by JetpackJackson** [comment]
Ok, I was able to successfully build on WSL and sign and install with AppManager. After importing the dictionaries that I wanted (English and German), it seems that suggestions are incompatible with glide typing; if I have both on, glide typing doesn't work, it does the "trail" but no words are typed. If I turn off glide typing and type each letter one by one, then I get suggestions. Is this a bug or an intended feature? I can understand if there's some issue with the code for glide typing not playing nice with the code for suggestions (finger/key detection maybe?). But other than that, it looks like suggestions work. I'm going to see if there's a way to get it to work with glide typing, since I use that often.

**Comment by Doc-4** [comment]
I turned on glide typing as a test before. It started working but I don't know what I had done to get it working

**Comment by JetpackJackson** [comment]
Glide typing with suggestions you mean?

**Comment by Doc-4** [comment]
No, just glide typing in general

**Comment by JetpackJackson** [comment]
Huh. Interesting. It hasn't done that for me (yet). I made sure to build the right branch, I'm not sure how else I could try and debug it.

**Comment by seniorm0ment** [comment]
@nettnikl I just installed using Obtanium (pre release selected) 0.4.0v4
They do not seem to be showing word suggestions still.

**Comment by Doc-4** [comment]
> @nettnikl I just installed using Obtanium (pre release selected) 0.4.0v4 They do not seem to be showing word suggestions still.

Obtainium doesn't build apps, it just fetches apps. If you want suggestions currently you'll need to build it yourself

**Comment by itsmartashub** [comment]
@seniorm0ment There is no official pre release with word suggestion implemented yet. Tho, you can try the `debug` version from GitHub Actions, which, it seems, has word suggestion. Download `app-debug.apk` from [here](https://github.com/florisboard/florisboard/actions/runs/7148610406), unzip, and install the `apk`.

**Comment by Doc-4** [comment]
> @seniorm0ment There is no official pre release with word suggestion implemented yet. Tho, you can try the `debug` version from GitHub Actions, which, it seems, has word suggestion. Download `app-debug.apk` from [here](https://github.com/florisboard/florisboard/actions/runs/7148610406), unzip, and install the `apk`.

If the debug apks generated by actions are the same as the ones assembleDebug creates, they won't work. Suggestions suggest placeholders in those

**Comment by itsmartashub** [comment]
@Doc-4 Hmm, I'm not sure when you built the app. It seems that these commits in GitHub Actions are from this morning (`18 hourse ago`), so maybe it's something different from what you built, idk, uh.
By the way, I'm using this version I mentioned, and autosuggestions work great (glide typing suggestions now working, or I'm doing something wrong). Ofc, with `en` dictionary downloaded from [here](https://github.com/florisboard/nlp/blob/main/data%2Fdicts%2Fv0%7Edraft1%2FREADME.md)

**Comment by Doc-4** [comment]
> @Doc-4 Hmm, I'm not sure when you built the app. It seems that these commits in GitHub Actions are from this morning (`18 hourse ago`), so maybe it's something different from what you built, idk, uh. By the way, I'm using this version I mentioned, and autosuggestions work great (glide typing suggestions now working tho, or I'm doing something wrong). Ofc, with `en` dictionary downloaded from [here](https://github.com/florisboard/nlp/blob/main/data%2Fdicts%2Fv0%7Edraft1%2FREADME.md)

Mine's working, that means it's not the same. Building the app with assembleDebug must get you a different result than github actions, then.

**Comment by patrickgold** [comment]
It's important to note that not all debug builds from GitHub actions contain NLP functionality, only those generated from the NLP implementation PR, and the newest debug APK from any branch is not from that PR.

**Comment by lm41** [comment]
> Howdy!
> 
> Nice keyboard, I love open source philosophy 💯 
> 
> Coming across this: How can I get suggestions to work? I'm on GrapheneOS on a Pixel 7a, so Android 15, in case it matters!

Hey. Currently you can't and it is a major thing planned for v0.6.

**Comment by rahulbansal16** [comment]
When is the v0.6 planned for launch?

**Comment by lm41** [comment]
There are no ETAs.
